src/rabbit-event-bus/amqp-connector.ts

```diff
@@ -161,6 +110,81 @@ export default class AMQPConnector {
             .getOrElse(false);
     }
 
+    public get subscribedEvents(): EventType[] {
+        return this.subscriptions;
+    }
+
+    private async createChannel(): Promise<Option<amqplib.Channel>> {
+        const conn = Option.of(this.connection);
+        if (conn.isEmpty()) {
+            return None;
+        } else {
+            return Option.of(await conn.get().createChannel());
+        }
+    }
+
+    private decoratedHandler<P extends object>(
+        rabbitChannel: amqplib.Channel,
+        handler: (ev: Event<P>) => Promise<boolean>,
+        msg: Message,
+    ): void {
+        try {
+            const message: EventBusMessage<Event<P>> = JSON.parse(msg.content.toString());
+
+            handler(message.event).then(isOk => {
+                if (isOk) {
+                    // Ack
+                    rabbitChannel.ack(msg);
+                } else {
+                    // Nack
+                    logger.warn('eventHandlerFailure');
+                    rabbitChannel.nack(msg, false, true);
+                }
+            });
+        } catch (e) {
+            rabbitChannel.nack(msg, false, false);
+            logger.warn(`Can't parse JSON! Error: ${e}`);
```

Comment:
Would it be useful to include any extra output to pinpoint this function as the source of the error? Incase we have other `JSON.parse` calls being try/catched in future?

Reply:
@Cooryd Done